Documentation/FAQ.md

```diff
@@ -0,0 +1,94 @@
+# Swift Protobuf FAQ
+
+This document is meant to cover some of the common/repeated questions in past
+issues/PRs.
+
+## Can you add an option for…
+
+While there are a [_few_ options that control the generated
+files](https://github.com/apple/swift-protobuf/blob/master/Documentation/PLUGIN.md#how-to-specify-code-generation-options),
+the intent is not to add customization points like this.
+
+The problem with generation options, especially if the change the generated PI
+surface, is it leads to problems where two pieces of code depend on the same
+`.proto` files. Once piece of code might decided they prefer the option set one
+way but the other might want it set differently. This means the two libraries
+can't share the generated code for the common .proto files.  Worse, they might
+not even be able to exist in the same library/binary.
+
+##  Why aren't there memberwise initializers?
+
+A `.proto` file does not consider the order the fields are listed in as
+important, they can be reorder, new fields add with updates in any order, etc.
+From a protocol buffers point of view, those aren't breaking changes as protocol
+buffers were design (especially the binary format) to support doing these things
+safely.
+
+If memberwise initializers are generated, then the problem becomes what order
```

Comment:
This creates problems for member wise initializers.  A large codebase taking advantage of such initializers would need widespread source changes if a new `.proto` definition changed the order of fields.  In addition, member wise initializers are impractical with very large messages.  Imposing an arbitrary limit on the size of such initializers would provide another avenue for source code breakage.

Reply:
Done